main/FAB/index.tsx

```diff
@@ -0,0 +1,127 @@
+import {Animated, Easing, StyleProp, View, ViewStyle} from 'react-native';
+import {ButtonSize, IconButton} from '../IconButton';
+import {DoobooTheme, withTheme} from '../theme';
+import {Icon, IconName} from '../Icon';
+import React, {ReactElement, useLayoutEffect, useMemo, useRef} from 'react';
+
+import styled from '@emotion/native';
+
+export const StyledIcon = styled(Icon)`
+  color: ${({theme}) => theme.textContrast};
+`;
+export interface FABItem {
+  icon: IconName;
+  id: string;
+}
+
+export interface FABProps<Item extends FABItem> {
+  isActive: boolean;
+  fabItems: Item[];
+  onPressFAB: () => void;
+  onPressFabItem: (item?: Item) => void;
+  renderFAB?: () => ReactElement;
+  renderFabItem?: (item: Item, idx: number) => ReactElement;
+  size: ButtonSize;
+  style?: StyleProp<ViewStyle>;
+}
+
+function FloatingActionButtons<Item extends FABItem = FABItem>({
+  isActive,
+  fabItems,
+  onPressFAB,
+  onPressFabItem,
+  renderFAB,
+  renderFabItem,
+  size = 'large',
+  style,
+}: FABProps<Item> & {
+  theme: DoobooTheme;
+}): ReactElement {
+  const spinValue = useRef(new Animated.Value(0));
+  const positionValue = useRef(new Animated.Value(0));
+
+  useLayoutEffect(() => {
+    Animated.timing(spinValue.current, {
+      toValue: isActive ? 1 : 0,
+      duration: 300,
+      easing: Easing.linear,
+      useNativeDriver: true,
+    }).start();
+  }, [isActive]);
+
+  useLayoutEffect(() => {
+    Animated.timing(positionValue.current, {
+      toValue: isActive ? 1 : 0,
+      duration: 300,
+      easing: Easing.linear,
+      useNativeDriver: true,
+    }).start();
+  }, [isActive]);
+
+  const spin = spinValue.current.interpolate({
+    inputRange: [0, 1],
+    outputRange: ['0deg', '45deg'],
+  });
```

Comment:
I think It would've been difficult to name it because `spinValue` was already defined.

I usually take this difficulty as a sign of changing the structure. 
I suggest you to inline this in `transform : rotate`.